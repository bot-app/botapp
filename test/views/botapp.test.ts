import { describe, expect, test } from "vitest";

import { botappView } from "../../src/views/botapp.js";

describe("botappView", () => {
  test("not providing parameters", () => {
    expect(botappView({})).toMatchSnapshot();
  });

  test("providing name", () => {
    expect(
      botappView({
        name: "My App",
      }),
    ).toMatchSnapshot();
  });

  test("providing description", () => {
    expect(
      botappView({
        description: "My App with Botapp",
      }),
    ).toMatchSnapshot();
  });

  test("providing description", () => {
    expect(
      botappView({
        version: "1.0.0",
      }),
    ).toMatchSnapshot();
  });
});
