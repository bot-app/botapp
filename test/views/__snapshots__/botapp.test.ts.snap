// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`botappView > not providing parameters 1`] = `
"<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Your App | built with Botapp</title>
    <link rel="icon" href="/botapp/static/botapp-head.png">
    <link rel="stylesheet" href="/botapp/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/botapp/static/robot.svg" alt="Botapp Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to your Botapp App
        </h1>

          <p>This bot was built using <a href="https://github.com/bot-app/botapp">Botapp</a>, a framework for building GitHub Apps.</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://botapp.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/bot-app/botapp/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`botappView > providing description 1`] = `
"<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Your App | built with Botapp</title>
    <link rel="icon" href="/botapp/static/botapp-head.png">
    <link rel="stylesheet" href="/botapp/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/botapp/static/robot.svg" alt="Botapp Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to your Botapp App
        </h1>

          <p>My App with Botapp</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://botapp.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/bot-app/botapp/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`botappView > providing description 2`] = `
"<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Your App | built with Botapp</title>
    <link rel="icon" href="/botapp/static/botapp-head.png">
    <link rel="stylesheet" href="/botapp/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/botapp/static/robot.svg" alt="Botapp Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to your Botapp App
            <span class="Label Label--outline v-align-middle ml-2 text-gray-light">v1.0.0</span>
        </h1>

          <p>This bot was built using <a href="https://github.com/bot-app/botapp">Botapp</a>, a framework for building GitHub Apps.</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://botapp.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/bot-app/botapp/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`botappView > providing name 1`] = `
"<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>My App | built with Botapp</title>
    <link rel="icon" href="/botapp/static/botapp-head.png">
    <link rel="stylesheet" href="/botapp/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/botapp/static/robot.svg" alt="Botapp Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to My App
        </h1>

          <p>This bot was built using <a href="https://github.com/bot-app/botapp">Botapp</a>, a framework for building GitHub Apps.</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://botapp.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/bot-app/botapp/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;
